Assets/Scripts/LetterBlock.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections;
+using UnityEngine;
+
+/// <summary>
+/// Choose letter block color, handle shake and break animation
+/// </summary>
+public class LetterBlock : MonoBehaviour
+{
+    [SerializeField] private GameObject letter;
+    [SerializeField] private GameObject particles;
+    [SerializeField] private float blockBreakDelay;
+    [SerializeField] [Range(0f, 0.1f)] private float delayBetweenShakes = 0f;
+    [SerializeField] [Range(0f, 2f)] private float distance = 0.1f;
+
+    private SpriteRenderer _renderer;
+    private Vector3 _startPos;
```

Comment:
Totally a style thing, but you could have this field be computed at the start of `Shake` and have it be local rather than a field